src/main.rs

```diff
@@ -367,19 +355,33 @@ fn main() {
     }
     #[cfg(windows)]
     {
+        use future::Either;
+        use futures::future::IntoFuture;
+
         let ctrl_c = tokio_signal::ctrl_c().flatten_stream().into_future();
         let crash = future::poll_fn(move || graceful_crash.poll());
 
-        let interruption = rt
-            .block_on(ctrl_c.select2(crash))
-            .map_err(|_| ())
-            .expect("Neither stream errors");
-
-        use futures::future::Either;
+        let mut ctrl_c_select = ctrl_c.select2(crash);
+        let mut change = future::poll_fn(move || config_watcher.poll()).into_future();
+        let ctrl_c = loop {
+            let interruption = rt
+                .block_on(ctrl_c_select.select2(change))
+                .map_err(|_| ())
+                .expect("Neither stream errors");
 
-        let ctrl_c = match interruption {
-            Either::A(((_, ctrl_c_stream), _)) => ctrl_c_stream.into_future(),
-            Either::B((_, ctrl_c)) => ctrl_c,
+            match interruption {
+                Either::A((interruption, _)) => {
+                    break match interruption {
```

Comment:
seems like a lot going on here can we beak up these lines?

Reply:
Something like this?